pyfifotax/utils.py

```diff
@@ -0,0 +1,667 @@
+        df_deposits["comment"] = ""
+        df_deposits.rename(
+            mapper={"fmv_or_buy_price": "fair_market_value"},
+            axis="columns",
+            inplace=True,
+        )
+        df_deposits = df_deposits[df_deposits.fees == 0]
+        df_deposits.drop(labels=["fees"], axis="columns", inplace=True)
+
+        df_buy_orders = df_buy_orders[df_buy_orders.fees != 0]
+        df_buy_orders["comment"] = ""
+        df_buy_orders.rename(
+            mapper={"fmv_or_buy_price": "buy_price", "net_quantity": "quantity"},
+            axis="columns",
+            inplace=True,
+        )
+
+        dtypes = {
+            "date": None,
+            "symbol": str,
+            "amount": np.float64,
+            "tax_withholding": np.float64,
+            "currency": str,
+        }
+        df_dividends = pd.read_excel(
+            xls, sheet_name="dividends", parse_dates=["date"], dtype=dtypes
+        )
+        df_dividends["comment"] = ""
+
+        dtypes = {
+            "date": None,
+            "symbol": str,
+            "quantity": np.float64,
+            "sell_price": np.float64,
+            "fees": np.float64,
+            "currency": str,
+        }
+        df_sell_orders = pd.read_excel(
+            xls, sheet_name="sales", parse_dates=["date"], dtype=dtypes
+        )
+        df_sell_orders["comment"] = ""
+
+        dtypes = {
+            "date": None,
+            "net_amount": np.float64,
+            "fees": np.float64,
+            "currency": str,
+        }
+        df_currency_conversions = pd.read_excel(
+            xls, sheet_name=forex_sheet, parse_dates=["date"], dtype=dtypes
+        )
+        df_currency_conversions["target_currency"] = "EUR"
+        df_currency_conversions["foreign_amount"] = (
+            df_currency_conversions.net_amount + df_currency_conversions.fees
+        )
+        df_currency_conversions.rename(
+            mapper={"fees": "source_fees", "currency": "source_currency"},
+            axis="columns",
+            inplace=True,
+        )
+        df_currency_conversions.drop(
+            labels=["net_amount"], axis="columns", inplace=True
+        )
+        df_currency_conversions["comment"] = ""
+
+        df_stock_splits = None
+        df_espp = None
+        df_currency_movements = None
+
+    return RawData(
+        df_deposits,
+        df_espp,
+        df_dividends,
+        df_buy_orders,
+        df_sell_orders,
+        df_currency_conversions,
+        df_currency_movements,
+        df_stock_splits,
+    )
+
+
+def read_data(sub_dir, file_name):
+    # read list of deposits, sales, dividend payments and currency conversions
+    # sort them to ensure that they are in chronological order
+    file_path = os.path.join(sub_dir, file_name)
+    with pd.ExcelFile(file_path) as xls:
+        dtypes = RSURow.type_dict()
+        df_rsu = pd.read_excel(
+            xls, sheet_name="rsu", parse_dates=["date"], dtype=dtypes
```

Comment:
Exception with:
```
"dtype '<class 'datetime.datetime'>' not understood (sheet: rsu)"
```
I think it should be `date.datetime`

Reply:
technically, datetime shouldn't be part of dtypes anyways because we explicitly also have `parse_dates`, I just forgot to set it to `None` for reading `df_rsu`